tools/java/org/unicode/cldr/test/ExampleGenerator.java

```diff
@@ -167,44 +178,157 @@ public CLDRFile getCldrFile() {
          */
+        if (AVOID_CLEARING_CACHE) {
+            if (!pathMightBeTypeA(xpath)) {
+                return;
+            }
+        }
         cache.clear();
     }
 
+    static long typeACount = 0, notTypeACount = 0;
+
+    /**
+     * Does changing the winning value for the given path potentially have side-effect of changing the example
+     * html for other paths? In other words, might this be a path of type "A"?
+     *
+     * We say "might be", since this function is meant to be fast rather than exact. It should never return
+     * false for a path that really is type "A", but it may return true for some paths that aren't really type "A".
+     *
+     * @param xpath
+     * @return true or false
+     *
+     * Called locally (only if AVOID_CLEARING_CACHE is true), and also by TestExampleGeneratorDependencies.
+     */
+    static public boolean pathMightBeTypeA(String xpath) {
+        final String pathAStarts[] = {
+            "//ldml/characterLabels/characterLabelPattern",
+            "//ldml/characterLabels/characterLabel",
+            "//ldml/dates/calendars",
+            "//ldml/dates/fields",
+            "//ldml/dates/timeZoneNames",
+            "//ldml/delimiters/alternateQuotationEnd",
+            "//ldml/delimiters/alternateQuotationStart",
+            "//ldml/delimiters/quotationEnd",
+            "//ldml/delimiters/quotationStart",
+            "//ldml/listPatterns/listPattern",
+            "//ldml/localeDisplayNames/codePatterns",
+            "//ldml/localeDisplayNames/keys",
+            "//ldml/localeDisplayNames/languages",
+            "//ldml/localeDisplayNames/localeDisplayPattern",
+            "//ldml/localeDisplayNames/scripts",
+            "//ldml/localeDisplayNames/territories",
+            "//ldml/localeDisplayNames/types",
+            "//ldml/numbers/currencies",
+            "//ldml/numbers/currencyFormats",
+            "//ldml/numbers/decimalFormats",
+            "//ldml/numbers/defaultNumberingSystem",
+            "//ldml/numbers/minimalPairs",
+            "//ldml/numbers/minimumGroupingDigits",
+            "//ldml/numbers/miscPatterns",
+            "//ldml/numbers/otherNumberingSystems",
+            "//ldml/numbers/percentFormats",
+            "//ldml/numbers/scientificFormats",
+            "//ldml/numbers/symbols",
+            "//ldml/posix/messages",
+            "//ldml/typographicNames",
+            "//ldml/units/durationUnit",
+            "//ldml/units/unitLength",
+        };
+        /***
+        final String pathAStartsShorter[] = {
+            "//ldml/characterLabels",
+            "//ldml/dates",
+            "//ldml/delimiters",
+            "//ldml/listPatterns",
+            "//ldml/localeDisplayNames",
+            "//ldml/numbers",
+            "//ldml/units"
+        };
+        ***/
+        boolean maybeTypeA = false;
+        for (String s : pathAStarts) {
+            if (xpath.startsWith(s)) {
```

Comment:
should use a hash

Reply:
Good idea, if the function continues to exist. I was hoping to find simple patterns for quick separation between most paths that have dependencies and most paths that don't. Disappointingly, there don't seem to be any such patterns. Instead, Mark and I are planning to build the complete mappings based on starred paths, definitely using a fast lookup (e.g., a hash or multimap), not looping through an array.